### PR TITLE
added leetcode solution for Candy

### LINKED ISSUE #683: add solution to leetcode problem 135 : Candy
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue Description

solution to leetcode problem 135

### Suggested Change

.

### Rationale

_No response_

### Urgency

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by mahek0620**
## Related Issue

Closes #683 

## Description

keetcode problem 135 solution added

## Type of PR

- [ ] Bug fix
- [X] Feature enhancement
- [ ] Documentation update
- [ ] Other (specify): _______________

## Checklist
- [X] I have performed a self-review of my code.
- [X] I have read and followed the Contribution Guidelines.
- [X] I have tested the changes thoroughly before submitting this pull request.
- [X] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [X] I have commented my code, particularly in hard-to-understand areas.

### CODE REVIEW

File: dsa-solutions/lc-solutions/0100-0199/0135-Candy.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+Explanation: You can allocate to the first, second and third child with 1, 2, 1 candies respectively.
+The third child gets 1 candy because it satisfies the above two conditions.
+
+```
+Input: root = [1,2,2,null,3,null,3]
+Output: false
+```
+
+### Constraints
+
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0100-0199/0135-Candy.md:49]: **Bad Practice**

```md
- n == ratings.length
- 1 <= n <= 2 * 10^4
- 0 <= ratings[i] <= 2 * 10^4
```

**Good Practice**

```md
- $n == ratings.length$
- $1 <= n <= 2 * 10^4$
- $0 <= ratings[i] <= 2 * 10^4$
```

File: dsa-solutions/lc-solutions/0100-0199/0135-Candy.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        return totalCandies;
+    }
+};
+    ```
+  </TabItem>
+</Tabs>
+
+#### Complexity Analysis
+
+- Time Complexity: O(n) where n is the number of children (length of the ratings array).
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0100-0199/0135-Candy.md:181]: replace `O(n)` to `$O(n)$`

File: dsa-solutions/lc-solutions/0100-0199/0135-Candy.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+- Space Complexity: O(n) for storing the candies array.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0100-0199/0135-Candy.md:182]: replace `O(n)` to `$O(n)$`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by mahek0620** [comment]
@Ajay-Dhangar the code has been updated based on the suggestion

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Review by Ajay-Dhangar — APPROVED** [review]
done and fixed bug

**Comment by Ajay-Dhangar** [comment]
@mahek0620, to calculate the score for your leaderboard, you'll need to add something to those files and then try it out at the bottom. Below is the code you need to include for your author profile in the documentation:

```md
---

<h2>Authors:</h2>

<div style={{display: 'flex', flexWrap: 'wrap', justifyContent: 'space-between', gap: '10px'}}>
{['github_username'].map(username => (
 <Author key={username} username={username} />
))}
</div>
```

**Comment by mahek0620** [comment]
@Ajay-Dhangar Do i need to create new PR because this PR is closed I can't edit any file and also can you mention the files which I need to change

**Pull request merged**